osu.Game/Screens/Menu/ButtonSystem.cs

```diff
@@ -256,6 +339,96 @@ public void OnReleased(KeyBindingReleaseEvent<GlobalAction> e)
         {
         }
 
+        public void SelectNext(int d)
+        {
+            if (d > 0 && State == ButtonSystemState.Initial)
+            {
+                State = ButtonSystemState.TopLevel;
+                SelectionIndex = 1;
+            }
+            else if (CurrentButtonsList == null)
+            {
+                SelectionIndex = -1;
+            }
+            else
+            {
+                int nextSelection = SelectionIndex;
+                if (nextSelection == -1)
+                    nextSelection = d; // start at 1, to not focus back/settings button, except when clicking left (focuses on settings)
+                else
+                    nextSelection += d;
+
+                if (nextSelection >= CurrentButtonsList.Count)
+                    nextSelection = CurrentButtonsList.Count - 1;
+                else if (nextSelection < 0)
+                    nextSelection = 0;
+
+                SelectionIndex = nextSelection;
+            }
+        }
+
+        public void SelectUp()
+        {
+            // if we come from play, focus play button again
+            int parentSelection = State == ButtonSystemState.Play ? buttonsTopLevel.FindIndex(b => b == PlayButton) : -1;
+            goBack();
+            SelectionIndex = parentSelection;
+        }
+
+        public void ClearKeyboardSelection(HoverEvent e)
```

Comment:
Would much prefer written as

```csharp
        public void ClearKeyboardSelection(HoverEvent e)
        {
            SelectionIndex = -1;
            if (CurrentButtonsList == null)
                return;

            foreach (var btn in CurrentButtonsList)
            {
                if (btn != e.Target)
                {
                    btn.SimulateHoverLost();
                }
            }
        }
```

Similar applies to `PreferKeyboardNavigation()`.